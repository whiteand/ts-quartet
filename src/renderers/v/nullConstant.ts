import { ISchemaRenderer } from "../../types";
import { fromValueSchemaAlloc } from "./utils";

export const nullConstantRenderer: ISchemaRenderer<null> = fromValueSchemaAlloc(
  valueId => `${valueId} === null`,
  valueId => `${valueId} !== null`
);
