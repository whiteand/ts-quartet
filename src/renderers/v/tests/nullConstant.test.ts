import { nullConstantRenderer } from "../nullConstant";
import { snapshot } from "./snapshot";

describe("v constant number renderer", () => {
  test("v(null)", () => {
    snapshot(nullConstantRenderer, { schema: null });
  });
});
