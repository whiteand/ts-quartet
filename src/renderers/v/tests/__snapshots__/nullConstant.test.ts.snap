// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`v constant number renderer v(null) 1`] = `"_valueId === null"`;

exports[`v constant number renderer v(null) 2`] = `"_valueId !== null"`;

exports[`v constant number renderer v(null) 3`] = `"if (_valueId === null) return true;"`;

exports[`v constant number renderer v(null) 4`] = `"if (_valueId !== null) return false;"`;
